gems/pending/VMwareWebService/MiqVimVm.rb

```diff
@@ -878,11 +879,30 @@ def getScsiCandU
       all_unit_numbers = [*0..MAX_SCSI_DEVICES]
       available_units  = all_unit_numbers - populated_units
 
-      unit_number = available_units.sort.first
-      break
+      available_units.each do |unit|
+        scsi_units << [controller_key, unit]
+      end
+    end
+
+    return scsi_units
+  end # def get_available_scsi_units
+
+  def get_available_scsi_buses
+    scsi_controller_bus_numbers = [*0..MAX_SCSI_CONTROLLERS - 1]
+
+    devices = getProp("config.hardware")["config"]["hardware"]["device"]
+
+    scsi_controllers = devices.select { |dev| VIRTUAL_SCSI_CONTROLLERS.include?(dev.xsiType) }
+    scsi_controllers.each do |controller|
+      scsi_controller_bus_numbers -= [controller["busNumber"].to_i]
     end
 
-    return controller_key, unit_number
+    scsi_controller_bus_numbers
+  end
+
+  def getScsiCandU
```

Comment:
Suggest to rename this method to something more readable. Or consider to eliminate this to get_available_scsi_units.first